yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryExecutor.java

```diff
@@ -273,10 +272,10 @@ private static Object executeStatement(@Nonnull Statement s, @Nullable String q)
         return execResult ? s.getResultSet() : s.getUpdateCount();
     }
 
-    private void setParametersInPreparedStatement(@Nonnull RelationalPreparedStatement s, @Nonnull Connection connection) throws SQLException {
+    private void setParametersInPreparedStatement(@Nonnull RelationalPreparedStatement statement) throws SQLException {
         int counter = 1;
         for (var parameter : Objects.requireNonNull(parameters)) {
-            s.setObject(counter++, parameter.getSqlObject(connection));
+            statement.setObject(counter++, parameter.getSqlObject(statement.getConnection()));
```

Comment:
While this "locks" the connection to the statement, I wonder whether that's actually required by the standard, and whether we should support `createStruct` and `createArray` that span multiple connections.

Reply:
I think, based on my reading of the spec, it is allowed, or at least should be. But, you should also be able to take a `Struct` or `Array` from any other database, and a whole bunch of other things.
I also feel like you're doing something weird if you are using two different connections here.
So, I feel like keeping the `YamlConnection` interface simpler at this point is worth more than automatically testing that an array can be passed between connections, and that testing that would be better done in a focused suite of tests that don't hide any of the complexity of JDBC behind a yaml file.